UnityProject/Assets/Scripts/Objects/PushPull.cs

```diff
@@ -438,6 +461,7 @@ public bool TryPush( Vector2Int dir, float speed = Single.NaN )
 	}
 
 	public bool TryPredictivePush( Vector3Int from, Vector2Int dir, float speed = Single.NaN ) {
+		//how does this differ from CanPush
```

Comment:
this one has clientside checks, and one that you extracted to `CanPush` has serverside checks